smart-wallets/src/interfaces/IYieldDistributionToken.sol

```diff
@@ -5,15 +5,86 @@ import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
 
 interface IYieldDistributionToken is IERC20 {
 
-    function getCurrencyToken() external returns (IERC20 currencyToken);
+    /**
+     * @notice Emitted when yield is deposited into the YieldDistributionToken
+     * @param user Address of the user who deposited the yield
+     * @param currencyTokenAmount Amount of CurrencyToken deposited as yield
+     */
+    event Deposited(address indexed user, uint256 currencyTokenAmount);
+
+    /**
+     * @notice Emitted when yield is claimed by a user
+     * @param user Address of the user who claimed the yield
+     * @param currencyTokenAmount Amount of CurrencyToken claimed as yield
+     */
+    event YieldClaimed(address indexed user, uint256 currencyTokenAmount);
+
+    /**
+     * @notice Emitted when yield is accrued to a user's balance
+     * @param user Address of the user who accrued the yield
+     * @param currencyTokenAmount Amount of CurrencyToken accrued as yield
+     */
+    event YieldAccrued(address indexed user, uint256 currencyTokenAmount);
+
+    /// @notice Indicates a failure because a yield deposit is made in the same block as the last one
+    error DepositSameBlock();
+
+    /// @notice Indicates a failure because the transfer of CurrencyToken failed
```

Comment:
@param user Address of the user whose transfer failed
@param currencyTokenAmount Amount of CurrencyToken to be transferred